### PR TITLE
Add fuzz test

### PR DESCRIPTION
**Pull request by mohanson**
Follow [linked-list-allocator](https://github.com/rust-osdev/linked-list-allocator/tree/main/fuzz) and add fuzz test

### CODE REVIEW

File: fuzz/fuzz_targets/chaos.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+static mut FAST_HEAP: Heap<FAST_HEAP_SIZE> = Heap([0u8; FAST_HEAP_SIZE]);
+static mut HEAP: Heap<HEAP_SIZE> = Heap([0u8; HEAP_SIZE]);
+
+fuzz_target!(|data: (u16, u32, u8, Vec<Action>)| {
+    let (fast_heap_size, heap_size, leaf_size, action_list) = data;
+    let fast_size = std::cmp::max(
+        64,
+        std::cmp::min(fast_heap_size as usize & 0xc0, FAST_HEAP_SIZE),
+    );
+    let heap_size = std::cmp::max(256, std::cmp::min(heap_size as usize & 0xc0, HEAP_SIZE));
```
Comment (XuJiandong) [inline fuzz/fuzz_targets/chaos.rs:30]: The 256 is too small. Suggest a more large number, like 16K.

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: fuzz/fuzz_targets/chaos.rs (hunk 1 above)
Comment (mohanson) [reply fuzz/fuzz_targets/chaos.rs:30]: What is the reason for increasing its minimum value? Currently, the range of heap size is [256, 1024 * 1024], which will include the value of 16K.

Comment (XuJiandong) [reply fuzz/fuzz_targets/chaos.rs:30]: That's enough.

File: fuzz/fuzz_targets/chaos.rs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+#[repr(align(64))]
+struct Heap<const S: usize>([u8; S]);
+static mut FAST_HEAP: Heap<FAST_HEAP_SIZE> = Heap([0u8; FAST_HEAP_SIZE]);
+static mut HEAP: Heap<HEAP_SIZE> = Heap([0u8; HEAP_SIZE]);
+
+fuzz_target!(|data: (u16, u32, u8, Vec<Action>)| {
+    let (fast_heap_size, heap_size, leaf_size, action_list) = data;
+    let fast_size = std::cmp::max(
+        64,
+        std::cmp::min(fast_heap_size as usize & 0xc0, FAST_HEAP_SIZE),
```
Comment (jjyr) [inline fuzz/fuzz_targets/chaos.rs:28]: `FAST_HEAP_SIZE` is 32KB, but fast_size is only determined by the `& 0xc0`, so the heap size of fast_allocator is one of `64`, `128`, `192`. I guess there are something wrong?

Comment (mohanson) [reply fuzz/fuzz_targets/chaos.rs]: Yes, it was a bug. It has been fixed.

**Pull request merged**